qmule/test/org/qcmg/qmule/FastqToSamWithHeadersTest.java

```diff
@@ -0,0 +1,56 @@
+package org.qcmg.qmule;
+
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.fastq.FastqRecord;
+import htsjdk.samtools.util.FastqQualityFormat;
+import org.junit.Test;
+
+import static htsjdk.samtools.SAMUtils.MAX_PHRED_SCORE;
+import static org.junit.Assert.*;
+
+public class FastqToSamWithHeadersTest {
+
+    @Test
+    public void createSAMRecordNoAdditionalHeader() {
+        FastqRecord fqRec = new FastqRecord("basename","ACGT", "", "????");
+        SAMRecord samRec = FastqToSamWithHeaders.createSamRecord(null, "basename", fqRec, true, "A", FastqQualityFormat.Standard, 0, MAX_PHRED_SCORE);
+        assertEquals("basename", samRec.getReadName());
+        assertEquals("ACGT", samRec.getReadString());
+        assertEquals("????", samRec.getBaseQualityString());
+        assertTrue(samRec.getReadPairedFlag());
+        assertNull(samRec.getAttribute(FastqToSamWithHeaders.ZT_ATTRIBUTE));
+        assertNull(samRec.getAttribute(FastqToSamWithHeaders.ZH_ATTRIBUTE));
+
+        samRec = FastqToSamWithHeaders.createSamRecord(null, "basename", fqRec, false, "A", FastqQualityFormat.Standard, 0, MAX_PHRED_SCORE);
+        assertEquals("basename", samRec.getReadName());
+        assertEquals("ACGT", samRec.getReadString());
+        assertEquals("????", samRec.getBaseQualityString());
+        assertFalse(samRec.getReadPairedFlag());
+        assertNull(samRec.getAttribute(FastqToSamWithHeaders.ZT_ATTRIBUTE));
+        assertNull(samRec.getAttribute(FastqToSamWithHeaders.ZH_ATTRIBUTE));
+    }
+
+    @Test
+    public void createSAMRecordAdditionalHeaderNoTrimming() {
+        FastqRecord fqRec = new FastqRecord("basename 1.2.ACGTTGCA/1", "ACGT", "", "????");
```

Comment:
Might I suggest two small changes to this and the ones below...
1. make the read id `basename/1 1.2.ACGTTGCA` so the `/1` is in a more typical position for paired reads
2. Construct the read id out of named pieces e.g. `readId = basename + additionalHeader + " TB:" + trimmedBases` so that it's a bit clearer in the `assertEquals`-es that the expected pieces are being put into the expected places

Reply:
tick and tick